src/components/filter/Filter.tsx

```diff
@@ -1,130 +1,211 @@
 /* eslint-disable consistent-return */
-import {useHistory, Link} from 'react-router-dom';
-import React, {useState, useEffect} from 'react';
+import React,
+{useState,
+  useEffect}
+  from 'react';
 import './filter.css';
-import {Checkbox} from '@patternfly/react-core/dist/js/components';
-import store from '../redux/store';
+import {Checkbox,
+  Tooltip,
+  Button,
+} from '@patternfly/react-core/dist/js/components';
 import {API_URL} from '../../constants';
+import {InfoCircleIcon} from '@patternfly/react-icons';
+import store from '../redux/store';
+import {FETCH_TASK_SUCCESS} from '../redux/Actions/TaskActionType';
 const Filter: React.FC = (props:any) => {
-  const history = useHistory();
-  const tagsSet = new Set();
-  const [tags, setTags] = useState([]);
-  const [toggle, setToggle] =useState('seeMore');
-  const [i, setI] =useState(10);
+  const [, setTags] = useState();
+  const [status, setStatus]=useState();
+  const [max, setMax]=useState(18);
+  const [, setX]=useState(1); //
+  const [clear, setClear]=useState(' ');
+  const [show, setShow]=useState('seeMore');
+  const tagitem :any =[{id: '1000', value: 'task', isChecked: false},
+    {id: '1001', value: 'pipeline', isChecked: false},
+    {id: '1002', value: 'verified', isChecked: false}];
+  const taglist=(data:any) =>{
+    data.map((it:any ) =>
+      tagitem.push({id: String(it.id), value: it.name, isChecked: false}));
+    return tagitem;
+  };
   useEffect(() => {
-    const fetchData = async () => {
-      await fetch(`${API_URL}/tags`)
-          .then((res) => res.json())
-          .then((data) => setTags(data));
-    };
-    fetchData();
+    // const fetchData = async () => {
+    fetch(`${API_URL}/tags`)
+        .then((res) => res.json())
+        .then((data) => setTags(taglist(data)));
+    setStatus(tagitem);
+    // eslint-disable-next-line
   }, []);
-  // sorting tags in alphabetical order
-  tags.sort((a:any, b:any) =>
-  (a.name> b.name) ? 1 :
-  ((b.name > a.name) ? -1 : 0));
-  let tagArray:any=[];
-  // adding tags into array after click and display task based on tags
-  const displaytask = () => {
-    let str: string = '';
-    if (tagsSet.has('task') === true) {
-      tagsSet.delete('task');
-    }
-    if (tagsSet.has('pipelines') === true) {
-      tagsSet.clear();
-      tagsSet.add('pipelines');
-    }
-    tagArray = Array.from(tagsSet);
-    for (let i = 0; i < tagArray.length; i++) {
-      if (i === 0 && tagArray[i] !== 'task') {
-        str += '?tags=';
+  // / dummy function for refreshing component
+  const ddff=() =>{
```

Comment:
what does this do? and can we have a meaningful name?

Reply:
i forget to give proper function name here  i am calling dummy function that set hooks ,i did it for refreshing the checkbox status, i will try to fix it.